plugins/database/cassandra/cassandra.go

```diff
@@ -152,29 +164,80 @@ func (c *Cassandra) RevokeUser(ctx context.Context, statements dbplugin.Statemen
 	c.Lock()
 	defer c.Unlock()
 
+	statements = dbutil.StatementCompatibilityHelper(statements)
+
 	session, err := c.getConnection(ctx)
 	if err != nil {
 		return err
 	}
 
-	revocationCQL := statements.RevocationStatements
-	if revocationCQL == "" {
-		revocationCQL = defaultUserDeletionCQL
+	revocationCQL := statements.Revocation
+	if len(revocationCQL) == 0 {
+		revocationCQL = []string{defaultUserDeletionCQL}
 	}
 
 	var result *multierror.Error
-	for _, query := range strutil.ParseArbitraryStringSlice(revocationCQL, ";") {
-		query = strings.TrimSpace(query)
-		if len(query) == 0 {
-			continue
-		}
+	for _, stmt := range revocationCQL {
+		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+			query = strings.TrimSpace(query)
+			if len(query) == 0 {
+				continue
+			}
 
-		err := session.Query(dbutil.QueryHelper(query, map[string]string{
-			"username": username,
-		})).Exec()
+			err := session.Query(dbutil.QueryHelper(query, map[string]string{
+				"username": username,
+			})).Exec()
 
-		result = multierror.Append(result, err)
+			result = multierror.Append(result, err)
+		}
 	}
 
 	return result.ErrorOrNil()
 }
+
+// RotateRootCredentials is not currently supported on Cassandra
+func (c *Cassandra) RotateRootCredentials(ctx context.Context, statements []string) (map[string]interface{}, error) {
+	// Grab the lock
+	c.Lock()
+	defer c.Unlock()
+
+	session, err := c.getConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	rotateCQL := statements
+	if len(rotateCQL) == 0 {
+		rotateCQL = []string{defaultRootCredentialRotationCQL}
+	}
+
+	password, err := c.GeneratePassword()
+	if err != nil {
+		return nil, err
+	}
+
+	var result *multierror.Error
```

Comment:
I'm not sure continuing on with statements after one has errored is the right way to go. At best it won't cause harm, but I wonder if it could leave things rather inconsistent.